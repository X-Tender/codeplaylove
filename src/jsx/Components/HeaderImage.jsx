import React from 'react';
import PropTypes from 'prop-types';

const HeaderImage = ({ src, isPixelated, caption }) => (
	<div className="header-image">
		<img
			className={`header-image__img ${isPixelated ? 'header-image__img--pixelate' : ''}`}
			src={src}
		/>
		{caption && <span className="header-image__caption">{caption}</span>}
	</div>
);

HeaderImage.propTypes = {
	src: PropTypes.string.isRequired,
	isPixelated: PropTypes.bool,
	caption: PropTypes.string,
};

HeaderImage.defaultProps = {
	isPixelated: false,
	caption: null,
};

export default HeaderImage;
