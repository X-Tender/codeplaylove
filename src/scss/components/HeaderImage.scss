.header-image {
	width: 100%;
	height: 25vw;
	overflow: hidden;
	position: relative;

	&__img {
		size: 100%;
		object-fit: cover;

		&--pixelate {
			image-rendering: optimizeSpeed;
			image-rendering: -moz-crisp-edges;
			image-rendering: -o-crisp-edges;
			image-rendering: -webkit-optimize-contrast;
			image-rendering: pixelated;
			image-rendering: optimize-contrast;
			-ms-interpolation-mode: nearest-neighbor;
		}
	}

	&__caption {
		position: absolute;
		bottom: 0;
		right: 0;
		padding: 5px 10px 5px 15px;
		border-top-left-radius: 50px;
		background-color: #fff;
		display: none;

		@include sm {
			display: block;
		}
	}
}
