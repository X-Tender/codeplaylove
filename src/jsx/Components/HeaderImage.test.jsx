import React from 'react';
import { shallow } from 'enzyme';
import HeaderImage from './HeaderImage';

describe('HeaderImage', () => {
	const src = 'image.jpg';
	const isPixelated = true;
	const caption = 'This is a caption';

	const comp = shallow(<HeaderImage caption={caption} isPixelated={isPixelated} src={src} />);

	it('renders correctly', () => {
		expect(comp).toMatchSnapshot();
	});

	it('renders the correct props', () => {
		expect(comp.find('img.header-image__img').prop('src')).toEqual(src);
		expect(comp.find('img.header-image__img.header-image__img--pixelate').exists()).toBe(true);
		expect(comp.find('span.header-image__caption').exists()).toBe(true);
		expect(comp.find('span.header-image__caption').text()).toEqual(caption);
	});
});
