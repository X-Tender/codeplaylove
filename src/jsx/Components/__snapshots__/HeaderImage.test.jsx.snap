// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderImage not pixelated renders correctly 1`] = `
<div
  className="header-image"
>
  <img
    className="header-image__img "
    src="image.jpg"
  />
  <span
    className="header-image__caption"
  >
    This is a caption
  </span>
</div>
`;

exports[`HeaderImage pixelated renders correctly 1`] = `
<div
  className="header-image"
>
  <img
    className="header-image__img header-image__img--pixelate"
    src="image.jpg"
  />
  <span
    className="header-image__caption"
  >
    This is a caption
  </span>
</div>
`;
